cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/ExpressionHandler.kt

```diff
@@ -182,45 +182,83 @@ class ExpressionHandler(frontend: PythonLanguageFrontend) :
-                strCall.addArgument(handle(node.value))
-                strCall
-            }
-            formattedValConversionRepr -> {
-                newProblemExpression(
-                    "Cannot handle conversion '114: !r repr formatting', yet.",
-                    rawNode = node
-                )
-            }
-            formattedValConversionASCII -> {
-                newProblemExpression(
-                    "Cannot handle conversion '97: !a ascii formatting', yet.",
-                    rawNode = node
-                )
+        val formatSpec = node.format_spec?.let { handle(it) }
+        val valueExpression = handle(node.value)
+        val conversion =
+            when (node.conversion) {
+                formattedValConversionNoFormatting -> {
+                    // No formatting, just return the value.
+                    valueExpression
+                }
+                formattedValConversionString -> {
+                    // String representation. wrap in str() call.
+                    val strCall =
+                        newCallExpression(
+                                newReference("str", rawNode = node),
+                                "str",
+                                rawNode = node
+                            )
+                            .implicit()
+                    strCall.addArgument(valueExpression)
+                    strCall
+                }
+                formattedValConversionRepr -> {
+                    // String representation. wrap in repr() call.
+                    val reprCall =
+                        newCallExpression(
+                                newReference("repr", rawNode = node),
+                                "repr",
+                                rawNode = node
+                            )
+                            .implicit()
+                    reprCall.addArgument(valueExpression)
+                    reprCall
+                }
+                formattedValConversionASCII -> {
+                    // String representation. wrap in ascii() call.
+                    val asciiCall =
+                        newCallExpression(
+                                newReference("ascii", rawNode = node),
+                                "ascii",
+                                rawNode = node
+                            )
+                            .implicit()
+                    asciiCall.addArgument(handle(node.value))
+                    asciiCall
+                }
+                else ->
+                    newProblemExpression(
+                        "Cannot handle formatted value with conversion ${node.conversion} yet",
+                        rawNode = node
+                    )
             }
-            else ->
-                newProblemExpression(
-                    "Cannot handle formatted value with conversion ${node.conversion} yet",
+        if (formatSpec != null) {
```

Comment:
Probably good to add some error handling. The AST doc says this is a `JoinedStr` and we treat it like that. However, it'd be good to catch some errors here for the `format` call to make sense!?

Reply:
Not the correct place (above for handle).